### PR TITLE
CLI: list containers from rawx and meta2

### PR DESCRIPTION
**Pull request by kamel-rahim**
##### SUMMARY
Build a list of containers:
- hosted by a meta2 service,
- which have chunks stored on a specific rawx service.

Jira: R1904-21

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
- CLI

##### SDS VERSION
```
openio 5.0.0.0a1.dev52
```

### CODE REVIEW

File: oio/cli/admin/service_list.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            metavar='<marker>',
+            help='Marker for paging'
+        )
+        parser.add_argument(
+            '--limit',
+            metavar='<limit>',
+            type=int,
+            default=1000,
+            help='Limit the number of objects returned (1000 by default)'
+        )
```
Comment (fvennetier) [inline oio/cli/admin/service_list.py]: Service ID, `--limit`, `--marker`, `--no-paging` and `--prefix` can apply to both `meta2 list` and `rawx list` commands, thus they can be implemented in `ServiceListCommand` class.

Also, the help string for `--marker` should tell that this marker is not one of the items returned, but a specific marker printed by the tool, aside from the list.

File: oio/cli/admin/service_list.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+from cliff import lister
+from oio.common.exceptions import ClientException
+
+
+class ServiceListCommand(lister.Lister):
+    """
+    A command to display items of a specific service
+    """
+
+    reqid_prefix = 'ACLI-LST'
```
Comment (fvennetier) [inline oio/cli/admin/service_list.py]: This is not used.

File: oio/cli/admin/service_list.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            pass
+        return cid
+
+
+class RawxListContainer(ServiceListCommand):
+    """
+    Get the names of all containers which some chunks are stored in the
+    specified rawx
+    """
+    reqid_prefix = 'ACLI-RLC'
```
Comment (fvennetier) [inline oio/cli/admin/service_list.py]: This is not used.

File: oio/cli/admin/service_list.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    def take_action(self, parsed_args):
+        super(RawxListContainer, self).take_action(parsed_args)
+        return ('Name', 'Chunks'), self._list_containers(parsed_args.rawx_id)
+
+
+class Meta2ListContainer(ServiceListCommand):
+    """
+    Get the names of all containers which are in the specified meta2
+    """
+    reqid_prefix = 'ACLI-M2LC'
```
Comment (fvennetier) [inline oio/cli/admin/service_list.py]: This is not used.

File: oio/cli/admin/service_list.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        reqid = self.app.request_id(self.reqid_prefix)
+        for r in self.rdir.meta2_index_fetch_all(
+                meta2, prefix=prefix, reqid=reqid):
+            yield r['container_url']
+
+    def _list_containers(self, meta2, **kwargs):
+        reqid = self.app.request_id(self.reqid_prefix)
+        resp = self.rdir.meta2_index_fetch(meta2,  reqid=reqid, **kwargs)
+        if resp.get('truncated'):
+            self.log.info(
```
Comment (murlock) [inline oio/cli/admin/service_list.py]: It should be `self.logger` IMO

File: oio/cli/admin/service_list.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+                'Object listing has been truncated, next marker: %s',
+                resp.get('next_marker'))
```
Comment (murlock) [inline oio/cli/admin/service_list.py]: `rest.get('next_marker')` is `None` when I test on my side

File: setup.cfg [hunk 7]
```diff
@@ -156,6 +156,8 @@ openio.admin =
     object_check = oio.cli.admin.item_check:ObjectCheck
     object_locate = oio.cli.admin.item_locate:ObjectLocate
     proxy_decache = oio.cli.admin.service_decache:ProxyDecache
+    rawx_list_containers = oio.cli.admin.service_list:RawxListContainers
+    meta2_list_containers = oio.cli.admin.service_list:Meta2ListContainers
```
Comment (fvennetier) [inline setup.cfg]: R.I.P. alphabetical order :cry:

File: oio/cli/admin/service_list.py (hunk 1 above)
Comment (fvennetier) [reply oio/cli/admin/service_list.py]: Ok, for the first version of the tool, we chose not to implement them for rawx operations.

**Pull request merged**